### PR TITLE
feat(DDD): Add IGeneratesDomainEvents

### PR DESCRIPTION
**Pull request by zhenlei520**

### CODE REVIEW

File: src/Ddd/Masa.BuildingBlocks.Ddd.Domain/Entities/AggregateRoot.cs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 namespace Masa.BuildingBlocks.Ddd.Domain.Entities;
 
-public abstract class AggregateRoot : Entity, IAggregateRoot
+public abstract class AggregateRoot : Entity, IGeneratesDomainEvents
```
Comment (doddgu) [inline src/Ddd/Masa.BuildingBlocks.Ddd.Domain/Entities/AggregateRoot.cs:6]: Why remove the IAggregateRoot?

File: src/Ddd/Masa.BuildingBlocks.Ddd.Domain/Entities/IGeneratesDomainEvents.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.BuildingBlocks.Ddd.Domain.Entities;
+
+public interface IGeneratesDomainEvents
```
Comment (doddgu) [inline src/Ddd/Masa.BuildingBlocks.Ddd.Domain/Entities/IGeneratesDomainEvents.cs:6]: IGeneratesDomainEvents -> IGenerateDomainEvents

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Ddd/Masa.BuildingBlocks.Ddd.Domain/Entities/AggregateRoot.cs (hunk 1 above)
Comment (zhenlei520) [reply src/Ddd/Masa.BuildingBlocks.Ddd.Domain/Entities/AggregateRoot.cs]: Misoperation, accidentally removed

File: src/Ddd/Masa.BuildingBlocks.Ddd.Domain/Entities/IGeneratesDomainEvents.cs (hunk 2 above)
Comment (zhenlei520) [reply src/Ddd/Masa.BuildingBlocks.Ddd.Domain/Entities/IGeneratesDomainEvents.cs]: processed

**Pull request merged**